python/plugins/processing/algs/qgis/Dissolve.py

```diff
@@ -86,13 +86,18 @@ def processAlgorithm(self, progress):
                 if first:
                     attrs = inFeat.attributes()
                     tmpInGeom = QgsGeometry(inFeat.geometry())
-                    if tmpInGeom.isGeosEmpty() or not tmpInGeom.isGeosValid():
+                    if tmpInGeom.isGeosEmpty():
+                        continue
+                    errors = tmpInGeom.validateGeometry()
+                    if len(errors) != 0:
+                        for error in errors:
+                            print error.what()
```

Comment:
The output should be sent to the log instead of printing it. E.g. https://github.com/matakuka/QGIS/blob/patch-1/python/plugins/processing/algs/qgis/Intersection.py#L111